service/controller/resource/tccp/create.go

```diff
@@ -116,6 +125,11 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		}
 
+		updateTags, err := r.detection.ShouldUpdateTags(ctx, key.ClusterID(&cr), stackTags)
```

Comment:
What I would like to see is to use the interfaces we have. The logic should simply be encapsulated in `ShouldUpdate`. Nobody should have to care about something rather complicated like `if key.IsAlreadyCreatedCluster(cr) && updateTags`. All that matters is `if update`.

Reply:
Maybe this gets easier when we simply drop all the legacy master handling. All we need should be this at this point. 

```
if update {
	err = r.updateStack(ctx, cr)
	if err != nil {
		return microerror.Mask(err)
	}
}
```

Might be we still need the other code for older versions upgrading to the new version. @calvix can you confirm we can clean this up or do we need to keep the master handling in here?

Reply:
Alternatives for the current proposal. Let's maybe think about this for a moment and understand benefits and drawbacks. 

* We could maybe put the tags directly into the Cloud Formation stacks, if that is possible. 
* We could have a separate handler for the tag propagation.